src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1344,7 +1344,7 @@ protected override void HandleGotoRequestAsync(IRequestResponder<GotoArguments>
                         if (!m_threads.TryGetValue(responder.Arguments.ThreadId, out thread))
                             throw new AD7Exception("Unknown thread id: " + responder.Arguments.ThreadId.ToString(CultureInfo.InvariantCulture));
                     }
-                    BeforeContinue();
+                    //BeforeContinue();
                     builder.CheckHR(thread.SetNextStatement(null, gotoTarget));
```

Comment:
Is that ok?

Reply:
You *probably* want BeforeContinue. The other debug adapter that the VS debugger team owns currently doesn't clear our handle collections on SetNextStatement, but you are probably correct that it should. But we should check VS Code and make sure it is happy.

Reply:
Yeah idk technically it doesn't run the program but on the other hand the current line of code needs to be updated. The thread cache invalidation worked in VSCode.